### DISCUSSION TITLE
"No space left on device" error during pip install on Linux

### DISCUSSION THREAD

**Discussion by forestfoxx** [openai/whisper#488] [Q&A] [answered]
Got stuck while downloading nvidia-cudnn-cu11 - "could not install packages due to an OSError: [Errno 28] No space left on device".

I have a new system with quite a lot of space, and df -h output confirms this. What could cause the problem? I tried using the --cache-dir argument to point to another tmp directory but it didn't help.

All the preceding wheels downloaded just fine.

**Comment by forestfoxx** [comment, marked as answer]
Bumping this

**Reply by jongwook** [reply]
`nvidia-cudnn-cu11` is a pretty large wheel (716 MB) and gets even larger when unzipped. Can you post the full output of `df -h` as well as what you see when you run `python -c "import numpy; print(numpy.__file__)"`?

**Reply by forestfoxx** [reply]
$ python -c "import numpy; print(numpy.__file__)"

/usr/lib/python3/dist-packages/numpy/__init__.py
(underscores around init and file aren't shown due to formatting but whatever)

```
$ df -h
Filesystem               Size  Used Avail Use% Mounted on
udev                     7.7G     0  7.7G   0% /dev
tmpfs                    1.6G  1.7M  1.6G   1% /run
/dev/mapper/Debian-root   23G   15G  6.7G  70% /
tmpfs                    7.7G     0  7.7G   0% /dev/shm
tmpfs                    5.0M     0  5.0M   0% /run/lock
/dev/nvme0n1p2           456M  158M  273M  37% /boot
/dev/mapper/Debian-home  198G   85G  104G  45% /home
/dev/mapper/Debian-tmp   1.8G   88K  1.7G   1% /tmp
/dev/mapper/Debian-var   9.1G  5.7G  3.0G  66% /var
/dev/loop1                71M   71M     0 100% /snap/core22/275
/dev/loop0                62M   62M     0 100% /snap/core20/1611
/dev/loop2                64M   64M     0 100% /snap/core20/1623
/dev/loop3                73M   73M     0 100% /snap/core22/310
/dev/loop8               128K  128K     0 100% /snap/bare/5
/dev/nvme0n1p1           511M  144K  511M   1% /boot/efi
/dev/loop4               347M  347M     0 100% /snap/gnome-3-38-2004/119
/dev/loop7               115M  115M     0 100% /snap/core/13886
/dev/loop6               115M  115M     0 100% /snap/core/13741
/dev/loop5               347M  347M     0 100% /snap/gnome-3-38-2004/115
/dev/loop9                92M   92M     0 100% /snap/gtk-common-themes/1535
/dev/loop10              918M  918M     0 100% /snap/libreoffice/259
/dev/loop11              919M  919M     0 100% /snap/libreoffice/260
tmpfs                    1.6G   56K  1.6G   1% /run/user/1000


```

**Reply by jongwook** [reply]
I think pip is failing to use your `/home` partition and likely using `/tmp` somehow. I'd try manually installing the package by

```bash
wget https://files.pythonhosted.org/packages/a4/28/9405bcded6412377529250a39313150c4dfd2993c67dc743b9f586bd6e1e/nvidia_cudnn_cu11-8.6.0.163-py3-none-manylinux1_x86_64.whl
pip install nvidia*.whl
```

under your home directory, maybe appending `--user` in the pip command to make it installed under `/home` rather than `/usr`.

**Reply by forestfoxx** [reply]
It successfully downloaded and installed that way (with --user), but after that I ran pip install git+https://github.com/openai/whisper.git and it tried downloading this wheel once again, with the same error

**Reply by forestfoxx** [reply]
Maybe it's because the version you linked is a bit newer than the one downloaded as part of the git+...... command. 8.5.0.96 vs 8.6.0.193

**Reply by forestfoxx** [reply]
Indeed, whisper seems to have installed fine when I changed the wheel version to the older one. Thanks! I wonder why did the problem came up in the first place, though.

**Reply by jongwook** [reply]
Glad it worked out!
